tika-parsers/src/main/java/org/apache/tika/parser/pkg/RarParser.java

```diff
@@ -86,7 +86,7 @@ public void parse(InputStream stream, ContentHandler handler,
                         subFile = rar.getInputStream(header);
 
                         Metadata entrydata = PackageParser.handleEntryMetadata(
-                                header.getFileNameString(),
+                                header.getFileNameW().equals("")?header.getFileNameString():header.getFileNameW(),

```

Comment:
Hi,
this comment was more about code style: If you use equals to compare 2 things, it is always better to have the part that is static and/or final and for sure not null (in this case: the string literal) first. Then you don't eben have to think about if it could be null or not. If you check your code with PMD it will suggest this to you.